Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -196,177 +196,205 @@ namespace AzToolsFramework
-                    else
+                case AZ::Dpe::Nodes::PropertyEditor::Align::AlignCenter:
                     {
-                        if (layoutIndex == 0)
-                        {
-                            itemAt(layoutIndex)->setGeometry(itemGeometry);
-                        }
-                        else
-                        {
-                            auto* currItem = itemAt(layoutIndex);
-                            itemGeometry.setLeft(itemGeometry.right() + 1);
-                            itemGeometry.setRight(itemGeometry.left() + perItemWidth);
-                            if (attributes)
-                            {
-                                switch (attributes->m_alignment)
-                                {
-                                case AZ::Dpe::Nodes::PropertyEditor::Align::AlignLeft:
-                                    currItem->setAlignment(Qt::AlignLeft);
-                                    break;
-                                case AZ::Dpe::Nodes::PropertyEditor::Align::AlignCenter:
-                                    currItem->setAlignment(Qt::AlignCenter);
-                                    break;
-                                case AZ::Dpe::Nodes::PropertyEditor::Align::AlignRight:
-                                    currItem->setAlignment(Qt::AlignRight);
-                                    break;
-                                }
-                            }
-                            currItem->setGeometry(itemGeometry);
-                        }
-                        layoutIndex++;
+                        startSpacer = true;
+                        endSpacer = true;
+                        break;
+                    }
+                case AZ::Dpe::Nodes::PropertyEditor::Align::AlignRight:
+                    {
+                        startSpacer = true;
+                        break;
                     }
+                default:
+                    break;
                 }
             }
+
+            // If the current widget is the first widget of a column, create the shared column layout and add widgets to it.
+            // First widget always creates a column.
+            if (m_columnStarts.contains(domIndex) || isFirstColumn)
+            {
+                // Close previous column.
+                if (!isFirstColumn)
+                {
+                    CloseColumn(
+                        currentColumnLayout,
+                        itemGeometry,
+                        currentColumnCount,
+                        columnWidth,
+                        (currentColumnNonStretchedWidgetsCount == currentColumnWidgetsCount),
+                        startSpacer,
+                        endSpacer
+                    );
+                }
+
+                // Create new column.
+                currentColumnLayout = new QHBoxLayout;
```

Comment:
in the long run, we don't want any sub-layouts in this class. This is a layout, it should be able to do all its child placement itself.